File: app/src/main/java/com/softhink/single/MapsActivity.java
package com.softhink.single;

import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetBehavior;
import android.support.v4.app.FragmentActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.SupportMapFragment;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MapStyleOptions;
import com.google.android.gms.maps.model.MarkerOptions;

public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {

    private GoogleMap mMap;
    private ImageView swipeView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_maps);
        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
                .findFragmentById(R.id.map);
        mapFragment.getMapAsync(this);

        LinearLayout layoutBottomSheet = findViewById(R.id.bottom_sheet);
        final LinearLayout layout = findViewById(R.id.view1);
        swipeView = findViewById(R.id.swipeView);

        final BottomSheetBehavior bottomSheetBehavior = BottomSheetBehavior.from(layoutBottomSheet);
        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_DRAGGING);
        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);

        bottomSheetBehavior.setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
            @Override
            public void onStateChanged(@NonNull View bottomSheet, int state) {
//                System.out.println(state);
                switch (state){
                    case BottomSheetBehavior.STATE_EXPANDED:
                        layout.setVisibility(View.GONE);
                        break;

                    case BottomSheetBehavior.STATE_SETTLING:
                        layout.setVisibility(View.VISIBLE);
                        break;

                    case BottomSheetBehavior.STATE_COLLAPSED:
                        layout.setVisibility(View.VISIBLE);
                        break;
                }
            }

            @Override
            public void onSlide(@NonNull View bottomSheet, float slideOffset) {
                animateBottomSheetArrow(slideOffset);
            }
        });
    }

    private void animateBottomSheetArrow(float slideOffset) {
        swipeView.setRotation(slideOffset * 180);
    }


    /**
     * Manipulates the map once available.
     * This callback is triggered when the map is ready to be used.
     * This is where we can add markers or lines, add listeners or move the camera. In this case,
     * we just add a marker near Sydney, Australia.
     * If Google Play services is not installed on the device, the user will be prompted to install
     * it inside the SupportMapFragment. This method will only be triggered once the user has
     * installed Google Play services and returned to the app.
     */
    @Override
    public void onMapReady(GoogleMap googleMap) {
        mMap = googleMap;



        try {
            mMap.setMapStyle(MapStyleOptions.loadRawResourceStyle(this, R.raw.style_json));
        }catch (Exception e){
            Log.e("SingleApp", e.getMessage());
        }

        // Add a marker in Sydney and move the camera
        LatLng sydney = new LatLng(19.428405, -99.137316);
        mMap.addMarker(new MarkerOptions().position(sydney).title("Single"));
        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(sydney, 13));
    }
}

File: app/src/main/java/com/softhink/single/dashboard/MapsActivity.kt
package com.softhink.single.dashboard

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import androidx.core.view.isVisible
import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.SupportMapFragment
import com.google.android.gms.maps.model.*
import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.softhink.single.R
import com.softhink.single.SingleHeaderAdapter
import kotlinx.android.synthetic.main.activity_maps.*
import kotlinx.android.synthetic.main.botom_sheet.*


class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMarkerClickListener, GoogleMap.OnMapClickListener {

    private lateinit var mMap: GoogleMap

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_maps)
        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
        val mapFragment = supportFragmentManager
                .findFragmentById(R.id.map) as SupportMapFragment
        mapFragment.getMapAsync(this)

        val strings: MutableList<String> = mutableListOf("", "", "", "", "", "", "", "", "", "")
        val adapter = SingleHeaderAdapter(this, R.layout.single_header, strings)
        scrollView_header.setAdapter(adapter, strings)
        scrollView_header.setCenter(0, adapter)

        val bottomSheetBehavior = BottomSheetBehavior.from(singleBottomSheet)
        bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
        bottomSheetBehavior.state = BottomSheetBehavior.STATE_DRAGGING
        bottomSheetBehavior.state = BottomSheetBehavior.STATE_HIDDEN

        bottomSheetBehavior.setBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback(){
            override fun onSlide(bottomSheet: View, slideOffset: Float) {
                animateBottomSheetArrow(slideOffset)
            }

            override fun onStateChanged(bottomSheet: View, state: Int) {
                when(state){
                    BottomSheetBehavior.STATE_DRAGGING ->{
                    }

                    BottomSheetBehavior.STATE_SETTLING -> {
                    }

                    BottomSheetBehavior.STATE_EXPANDED -> {
                        matcher.visibility = View.GONE
                        val layoutParams = LinearLayout.LayoutParams(120, 120)
                        singleImage.layoutParams = layoutParams
                        view1.layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)
                    }

                    BottomSheetBehavior.STATE_COLLAPSED -> {
                        matcher.visibility = View.VISIBLE
                        singleImage.layoutParams = LinearLayout.LayoutParams(440, 440)
                        view1.layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, 700)
                    }

                    BottomSheetBehavior.STATE_HIDDEN -> {
                    }

                    BottomSheetBehavior.STATE_HALF_EXPANDED -> {
                    }
                }
            }
        })
    }

    private fun animateBottomSheetArrow(offset: Float) {
        val slideOffset = String.format("%.1f", offset).toFloat()
        swipeView.rotation = slideOffset * 180
    }

    override fun onMapReady(googleMap: GoogleMap) {
        mMap = googleMap
        mMap.setMapStyle(MapStyleOptions.loadRawResourceStyle(this, R.raw.style_json))

        val mexicoCity = LatLng(19.429507, -99.163656)

        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(mexicoCity, 12f))

        val markers = arrayOf(LatLng(19.427411, -99.166339), LatLng(19.427517, -99.129610), LatLng(19.419569, -99.140644),
                LatLng(19.421987, -99.175018), LatLng(19.398625, -99.160190))
        for (lats in markers){
            val marker = MarkerOptions()
                    .position(lats)
                    .icon(BitmapDescriptorFactory.fromResource(R.drawable.avatar_mini))

            mMap.addMarker(marker)
        }

        mMap.setOnMarkerClickListener(this)
        mMap.setOnMapClickListener(this)
    }

    override fun onMarkerClick(marker: Marker?): Boolean {
        scrollView_header.visibility = View.VISIBLE
        singleBottomSheet.visibility = View.VISIBLE
        return true
    }

    override fun onMapClick(p0: LatLng?) {
        if (scrollView_header.isVisible && singleBottomSheet.isVisible){
            scrollView_header.visibility = View.GONE
            singleBottomSheet.visibility = View.GONE
        }
    }
}
